x/interchainstaking/keeper/ibc_packet_handlers.go

```diff
@@ -189,7 +189,10 @@ func (k *Keeper) HandleAcknowledgement(ctx sdk.Context, packet channeltypes.Pack
 		case "/cosmos.staking.v1beta1.MsgDelegate":
 			// TODO: can we safely ignore this?
 			if !success {
-				return nil
+				if err := k.HandleFailedDelegate(ctx, src, packetData.Memo); err != nil {
+					return err
+				}
+				continue
```

Comment:
Note, we now `continue`, rather than `return nil` as we previously were able to assume that all `MsgDelegate` in a tx had failed, and no further action was required - but because we now decrement the waitgroup, we need to iterate through all failed messages.